File: backend/routes/serviceRequest.js
const express = require("express");
const router = express.Router();
const ServiceRequest = require("../models/ServiceRequest");
const authMiddleware = require("../middleware/authMiddleware");

router.post("/book", authMiddleware, async (req, res) => {
  try {
    const { serviceType, vehicleType, description } = req.body;

    const newRequest = new ServiceRequest({
      userId: req.user.id,
      serviceType,
      vehicleType,
      description,
    });

    await newRequest.save();

    res.json({
      success: true,
      message: "Service request submitted successfully",
    });

  } catch (error) {
    res.status(500).json({
      success: false,
      message: "Server error",
    });
  }
});

module.exports = router;

File: motobuddy/lib/models/service_request.dart
// lib/models/service_request.dart
class ServiceRequest {
  final String serviceType;
  final String vehicleType;
  final String description;
  final String? imagePath; // optional, if user uploads an image

  ServiceRequest({
    required this.serviceType,
    required this.vehicleType,
    required this.description,
    this.imagePath,
  });

  Map<String, dynamic> toJson() => {
        "serviceType": serviceType,
        "vehicleType": vehicleType,
        "description": description,
        "imagePath": imagePath,
      };
}
